engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -263,6 +263,47 @@ namespace dmGameObject
         return regist->m_DefaultInputStackCapacity;
     }
 
+    void AddDynamicResourceHash(HCollection hcollection, dmhash_t resource_hash)
+    {
+        Collection* collection = hcollection->m_Collection;
+        dmMutex::Lock(collection->m_Mutex);
+        if (collection->m_DynamicResources.Remaining() == 0)
+        {
+            collection->m_DynamicResources.OffsetCapacity(1);
+        }
+        collection->m_DynamicResources.Push(resource_hash);
+        dmMutex::Unlock(collection->m_Mutex);
+    }
+
+    void RemoveDynamicResourceHash(HCollection hcollection, dmhash_t resource_hash)
+    {
+        Collection* collection = hcollection->m_Collection;
+        dmMutex::Lock(collection->m_Mutex);
+        for (int i = 0; i < collection->m_DynamicResources.Size(); ++i)
+        {
+            if (collection->m_DynamicResources[i] == resource_hash)
+            {
+                collection->m_DynamicResources.EraseSwap(i);
+            }
+        }
+        dmMutex::Unlock(collection->m_Mutex);
+    }
+
+    static void ReleaseDynamicResources(Collection* collection)
```

Comment:
Called when collection is destroyed to automatically release any added resources, as it is implied that the collection owns the dynamic resources that are added to it.